great_expectations/execution_engine/split_and_sample/sqlalchemy_data_splitter.py

```diff
@@ -173,7 +173,7 @@ def split_on_date_parts(
         """
         self._validate_date_parts(date_parts)
 
-        date_parts: List[DatePart] = self._convert_date_parts(date_parts)
+        date_parts = self._convert_date_parts(date_parts)
```

Comment:
Same comment...we have more information that we could convey after calling this method (`date_parts` can no longer be `List[str]`).

Reply:
https://github.com/great-expectations/great_expectations/pull/6730/files#r1064677778